x-pack/plugins/task_manager/server/task_pool.ts

```diff
@@ -181,7 +181,8 @@ export class TaskPool {
   }
 
   private handleMarkAsRunning(taskRunner: TaskRunner) {
-    taskRunner
+    // the `.then` at the end confuses TS and thinks it's an unhandled promise.
+    void taskRunner
```

Comment:
This one is a bit tricky:
`taskRunner.run()` may reject, but it's handled by `.catch`, which is mostly logging, and shouldn't throw unless there's a bug in the code (trying to access a property that does not exist or anything like that).

The ESlint checks that structure on its own and is OK with it. However, it identifies the following `.then` as a potential rejection threat. I'm using `void` because that `then` only deletes an element from a `Map`. It's almost impossible that it'll fail in any way.

That's why I used `void`: to tell the ESLint _trust me! I'm leaving this unhandled. I know what I'm doing_.

If you think it's best to add another catch at the end or to rewrite this method, let me know.

Reply:
Heh, ya, I think the `void` thing is going to be confusing to people - was to me :-)

I'd prefer we just always use `.catch()` instead in these cases.  Will save us in the future if the functionality expands and throwing becomes possible - seems like it could be easy to miss.